scrapy/core/scheduler.py

```diff
@@ -16,17 +17,75 @@
 logger = logging.getLogger(__name__)
 
 
-class Scheduler:
+class BaseScheduler:
```

Comment:
I think we should either raise `NotImplementedError` or mark as @abstractmethod those methods that do not simply work as implemented here (all except `open` and `close`).